### DISCUSSION TITLE
Consider making it so the CoC and Terms Links on /report-abuse are customizable

### DISCUSSION THREAD

**Discussion by michael-tharrington** [forem/forem#16117] [Feature Requests]
Right now, the default `/report-abuse` page links to the community's `/code-of-conduct` and `terms` pages.

Upon launching a Forem, it comes loaded with default `/code-of-conduct` and `terms` pages set up, so typically these links will work just fine.

However, if a Forem Creator wants to link to an external Code of Conduct (or Terms, which admittedly I don't know if that's even feasible... thinking a terms page would be required) then they have no way to change the links on this page.

**Comment by benhalpern** [comment]
I believe the right call is this:

Allow pages to _redirect-to_ any URL. It just becomes a new functionality of the `Page` model, but it could enable a lot of future functionality as well.

One such feature:

I think we could add pages to the search engine (and have it be optional whether or not they are discoverable that way — plenty of page uses you wouldn't want showing up in internal search)... And once we enable the Pages API, we can allow creators to mirror their official documentation for searchability within their Forems — but if they don't actually want to mirror them as pages within the Forems (though that is not such a bad idea), then the mirroring is only for exposing the _content_ to the search engine. The actual functionality is the redirect.

That's a digression, but getting back to the original point: I believe a `redirect_to` functionality for `Page` would solve this problem in a really clean way.

It should be a `301` redirect. Implementation of all this should be pretty straightforward.

**Reply by michael-tharrington** [reply]
Oh awesome! This is an excellent idea. 🙌

I could absolutely see other uses for this.

**Reply by benhalpern** [reply]
And it really is only a few lines of code and follows similar patterns of pages being a generally flexible building block.

My one meta comment is that the more we use pages, the more important sorting/searching/filtering will be on the admin panel. If, based on my train of thought, there wind up being 10k page objects in existence for a Forem, that's not a huge number in computer, but it could make finding basic key pages like `about` a lot harder.

**Reply by michael-tharrington** [reply]
Ooof, you're right about that. 😅
